src/components/TransactionViewPage/TransactionViewPage.tsx

```diff
@@ -1,45 +1,63 @@
 import React, { Component } from 'react';
 import './TransactionViewPage.scss';
 import Sidebar from '../TransactionViewSidebar/TransactionViewSidebar';
-import { Accordion, AccordionItem, Link } from 'carbon-components-react';
+import { Accordion, AccordionItem } from 'carbon-components-react';
 
 type PageProps = {
-    activeSmartContract: string
+    smartContracts: string[],
+    activeSmartContract: string,
+    switchSmartContract: (newSmartContract: string) => void
 };
 
 type PageState = { 
-    activeSmartContract: string
+    smartContracts: string[],
+    activeSmartContract: string,
+    switchSmartContract: (newSmartContract: string) => void
 }
 
 class TransactionViewPage extends Component<PageProps, PageState> {
 
     constructor(props: Readonly<PageProps>) {
         super(props);
         this.state = {
-            activeSmartContract: this.props.activeSmartContract
+            smartContracts: this.props.smartContracts,
+            activeSmartContract: this.props.activeSmartContract,
+            switchSmartContract: this.props.switchSmartContract
         }
     }
 
+    getSmartContractItems(smartContracts: string[]): any {
+        let smartContractItems: any = [];
```

Comment:
The type should probably be `any[]`, or even better... whatever the type of the <li> thing that you push to the array.